pkg/trace/api/api.go

```diff
@@ -148,8 +148,10 @@ func (r *HTTPReceiver) Start() {
 	}
 	go func() {
 		defer watchdog.LogOnPanic()
-		r.server.Serve(ln)
-		ln.Close()
+		if ln != nil {
```

Comment:
I think we prefer a panic here, rather than a silent "successful" startup without a server to receive anything and without any error logged. `ln` will never be `nil` outside tests. Please undo this code and let's not add workarounds for tests.

Any error with creating the listener should be caught by the error check above and `killProcess` is responsible for exiting.

Same comment applies below.

Reply:
I have a different point of view. As `killProcess` may not exit the process in some cases, the remaining part of the function should not panic because of a nil pointer exception. It is not a workaround for tests from my perspective.